### PR TITLE
Delay chunk unloads

### PR DESCRIPTION
**Pull request by Spottedleaf**
Chunk unloads are now delayed by a configurable amount.
Specifically, ticket level reduction is delayed by the configured amount.
This is done to allow players to teleport and have their pets follow them,
as the chunks will no longer unload or have entity ticking status removed.

It's also targetted to reduce performance regressions when
plugins or edge cases in code do not spam sync loads since chunks
without tickets get unloaded immediately.

### CODE REVIEW

File: Spigot-Server-Patches/0422-Delay-chunk-unloads.patch [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
++        }
++        boolean changed = tickets.removeIf((Ticket<?> ticket) -> {
++            return ticket.getTicketType() == type;
++        });
++        if (changed) {
++            this.getTicketTracker().update(chunk, getLowestTicketLevel(tickets), false);
++        }
++    }
++
++    private final java.util.function.LongFunction<Ticket<Long>> computeFuntion = (long key) -> {
```
Comment (electronicboy) [inline Spigot-Server-Patches/0422-Delay-chunk-unloads.patch]: This probably needs a better name?

**Review by electronicboy — APPROVED** [review]

**Comment by electronicboy** [comment]
Approved, just needs a quick change, might also be worth maaaybe clarifying that isCached field a bit more as to what it's tryna represent, but otherwise looks fine

> fike

File: Spigot-Server-Patches/0422-Delay-chunk-unloads.patch (hunk 1 above)
Comment (Spottedleaf) [reply Spigot-Server-Patches/0422-Delay-chunk-unloads.patch]: It's not even spelled correctly, looks like I meant to name it but forgot

**Comment by Spottedleaf** [comment]
hits too many areas of chunkloading, will be attached to view distance patch to only handle player chunks

**Pull request closed without merging**